source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,316 @@
+   ASPECT is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2, or (at your option)
+   any later version.
+
+   ASPECT is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with ASPECT; see the file doc/COPYING.  If not see
+   <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/initial_conditions/interface.h>
+#include <aspect/initial_conditions/adiabatic_boundary.h>
+#include <fstream>
+#include <iostream>
+
+/** This initial condition is designed only for the 3D ellipsoid chunk geometry.
+ */
+
+
+namespace aspect
+{
+  namespace InitialConditions
+  {
+    using namespace dealii;
+
+    /**
+     * Function that return latitude and longitude from cartesian
+     * coordinates that account for ellipsoidal shape of the Earth
+     */
+    template <int dim>
+    std::pair<double, double>
+    AdiabaticBoundary<dim>::lat_long_from_xyz_wgs84(const Point<3> &pos) const
+    {
+    	    /* WGS84 ellipsoid constants */
+    	    const double radius = 6378137;
+    	    const double ellipticity = 8.1819190842622e-2;
+
+    	    const double b = std::sqrt(radius * radius
+    	                               * (1 - ellipticity * ellipticity));
+    	    const double ep = std::sqrt((radius * radius - b * b) / (b * b));
+    	    const double p = std::sqrt(pos(0) * pos(0) + pos(1) * pos(1));
+    	    const double th = std::atan2(radius * pos(2), b * p);
+    	    const double lon = std::atan2(pos(1), pos(0));
+    	    const double lat = std::atan2((pos(2) + ep * ep * b * std::sin(th)
+    	                                   * std::sin(th) * std::sin(th)),
+    	                                  (p - (ellipticity
+    	                                        * ellipticity
+    	                                        * radius
+    	                                        * (std::cos(th) * std::cos(th)
+    	                                        * std::cos(th)))));
+
+    	    /* convert to degrees */
+    	    const double lon_degrees = lon * (180 / numbers::PI);
+    	    const double lat_degrees = lat * (180 / numbers::PI);
+
+    	    /* set all longitudes between [0,360] */
+    	    if (lon_degrees < 0)
+    	      return std::make_pair(lat_degrees, lon_degrees + 360);
+    	    else if (lon_degrees > 360)
+    	      return std::make_pair(lat_degrees, lon_degrees - 360);
+    	    else
+    	      return std::make_pair(lat_degrees, lon_degrees);
+
+    }
+
+    template <>
+    double
+    AdiabaticBoundary<2>::get_isotherm_depth(const double,
+                                              const double) const
+    {       abort();
```

Comment:
Replace the abort by `AssertThrow(false,ExcNotImplemented());` and put it on a line of its own.